### ISSUE TITLE
Triangle inequality?

### ISSUE THREAD

**Issue by cprieto** [exercism/go#1371]
Maybe I am super wrong here, but I remember the theorem of triangle inequality:

````
a + b > c
a + c > b
b + c > a
````

This means the triangles in the test:

- Iso, 2, 4, 2
- Sca, 1, 4, 3

Will fail triangle inequality

**Comment by tehsphinx** [comment]
The third point is on the line between the first and second point. You could view them as an edge case of a triangle (also called `degenerate triangles`)

Wikipedia:
> Three given angles form a non-degenerate triangle (and indeed an infinitude of them) if and only if both of these conditions hold: (a) each of the angles is positive, and (b) the angles sum to 180°. If degenerate triangles are permitted, angles of 0° are permitted.

With the two triangles you mention, we have the case of 2 angles being 0 degrees and one having 180 degrees.

Not that the triangle inequality will not hold up for `degenerate triangles` and would have to be expanded to be:

    a + b >= c
    a + c >= b
    b + c >= a

**Comment by dhavalgshah** [comment]
I came across the same and found the README and test cases do diverge. Also, this whole exercise in Go track is diverged from common problem specifications.

As per [problem-specifications#202](https://github.com/exercism/problem-specifications/issues/202) **degenerate triangles** test cases should be removed as they are overly complex for this exercise.

I checked [PHP](https://github.com/exercism/php/tree/main/exercises/practice/triangle), [Javascript](https://github.com/exercism/javascript/tree/main/exercises/practice/triangle), and [Erlang](https://github.com/exercism/erlang/tree/main/exercises/practice/triangle) tracks. None of them are having these test cases. It is more confusing because the README says to implement own test cases for degenerate triangles if you want to dig deeper.

I am also opening a separate issue as there are tests for `Inf+` lengths, which are neither in common specifications nor in any other tracks AFAIK.

@tehsphinx Either the README should be updated to mention that tests includes degenerate triangles as well as `Inf+` as lengths or this exercise should be simplifies to align with core specification. In case of former, this exercise should not be at its current level of required proficiency.
